### DISCUSSION TITLE
Problem with number format

### DISCUSSION THREAD

**Discussion by MichaelLoibl** [evalf/nutils#889] [Q&A] [answered]
Dear all,
I tried to define an immersed model with a function which I got from another program. However, directly using this function did not work. The function is (main version):
`domain = domain0.trim(-x/2 + y/36028797018963968, maxrefine=maxrefine)`
Which is equivalent to (alternative version):
`domain = domain0.trim(-x/2 + y*2.7755575615628914e-17, maxrefine=maxrefine)`
The alternative version works without problems. For the main version, which is the one that I get from the other program, I get the following error:
`RuntimeWarning: divide by zero encountered in reciprocal
  In C:\Users\Loibl\AppData\Local\Programs\Python\Python310\lib\site-packages\nutils\evaluable.py:370       
trimming 100% > RuntimeWarning: invalid value encountered in matmul
  In C:\Users\Loibl\AppData\Local\Programs\Python\Python310\lib\site-packages\nutils\element.py:204
AssertionError`

Do you know why this is happening? Is there a possibility to change the expression before I give it to the "trim"-method?

My code looks for example as follows:

 ```
domain0, geom = mesh.rectilinear([numpy.linspace(0, 1, 2), numpy.linspace(0, 1, 2)])

 x,y = geom
 x = (x*1.02)-0.51
 y = (y*1.02)-0.51
    
 domain = domain0.trim(-x/2 + y*2.7755575615628914e-17, maxrefine=maxrefine)
 domain = domain0.trim(-x/2 + y/36028797018963968, maxrefine=maxrefine)
```
    
Thank you for your help.
Best,
Michael

**Comment by joostvanzwieten** [comment, marked as answer]
Hi Michael,

There are two problems here. This first is that you're using a 32 bit version of Python and the number 36028797018963968 cannot be represented by a 32 bit signed integer. The second problem is Nutils not telling you this, but truncating the number to the least significant 32 bits silently, which is 0 in this particular case. The second problem is fixed in master (#890). The first problem is fixed by appending a dot, which makes it a floating point number, reducing the number to something that can be represented by a 32 bit signed integer, or switching to a 64 bit build of Python (which is the default on all platforms but Windows).

Best, Joost

**Reply by MichaelLoibl** [reply]
Dear Joost,

that was very helpful. It turned out to be even a bit more confusing in my case because I was already using a 64 bit Python. However, the default numpy integer type was 32 bit on Windows (and only on Windows) for older numpy releases. That was changed with the numpy release 2.0.0. (https://numpy.org/devdocs/release/2.0.0-notes.html). However, I used an older version on my computer.

Best, Michael
